File: GPPG/management/commands/watch_files.py
# myapp/management/commands/watch_files.py
from django.core.management.base import BaseCommand
import subprocess

class Command(BaseCommand):
    help = 'Watch files using chokidar'

    def handle(self, *args, **kwargs):
        self.stdout.write("Starting file watcher...")
        try:
            # Call chokidar watcher here, subprocess can run it
            subprocess.run(["node", "watch_files.js"], check=True)
        except subprocess.CalledProcessError as e:
            self.stderr.write(f"File watcher failed: {e}")
        except Exception as e:
            self.stderr.write(f"Unexpected error: {e}")

File: GPPG/management/commands/watch_files.js
const chokidar = require("chokidar");

const watcher = chokidar.watch(".", {
  ignored: /DumpStack\.log\.tmp/, // Ignore system file
  persistent: true,
});

watcher.on("all", (event, path) => {
  console.log(event, path);
});

watcher.on("error", (error) => {
  console.error("Error occurred:", error);
});